File: mall-ware/src/main/java/com/hj/mall/ware/entity/StockLog.java
package com.hj.mall.ware.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("wms_stock_log")
public class StockLog {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long wareId;

    private Long skuId;

    private String changeType;

    private Integer changeAmount;

    private Integer beforeStock;

    private Integer afterStock;

    private Integer beforeLocked;

    private Integer afterLocked;

    private String orderSn;

    private Long purchaseId;

    private String remark;

    private LocalDateTime createTime;
}

File: mall-admin/src/api/ware/stockLog.ts
import { get } from '../request'
import type { PageResult } from '../request'

export interface StockLog {
  id?: number
  wareId?: number
  wareName?: string
  skuId?: number
  skuName?: string
  type: number // 1: 入库, 2: 出库, 3: 调整
  quantity: number
  beforeStock: number
  afterStock: number
  remark?: string
  createTime?: string
}

export interface StockLogParams {
  key?: string
  wareId?: number
  skuId?: number
  type?: number
}

// 获取库存日志列表
export const getStockLogList = (params?: StockLogParams) => {
  return get<PageResult<StockLog>>('/admin/ware/stockLog/list', params)
}
